boards/arm/rpi_pico/doc/index.rst

```diff
@@ -0,0 +1,59 @@
+.. _rpi_pico:
+
+Raspberry Pi Pico
+#################
+
+Overview
+********
+
+The Raspberry Pi Pico is a small, low-cost, versatile board from
+Raspberry Pi. It is equipped with an RP2040 SoC, an on-board LED,
+a USB connector, and an SWD interface. The USB bootloader allows it
+to be flashed without any adapter, in a drag-and-drop manner.
+It is also possible to flash and debug the Pico with its SWD interface,
+using an external adapter.
```

Comment:
Could you please be a bit more descriptive here ?

Would it be possible to have `west flash` support for common tools like openocd.

As minimum, would be nice to have the flash description from here https://github.com/zephyrproject-rtos/zephyr/pull/34835#issue-875560798 included in this guide.

Reply:
Yes, I'll elaborate on this.

Regarding west flash:
Flashing with jlink requires a specific branch of a specific fork of openocd, so west can't rely on it unless we add it to the Zephyr setup.
With UF2, it's usually flashed with drag-and-drop to the "mass storage device", and should probably work with `cp zephyr.uf2 /media/...` (can west flash run arbitrary commands, or can it only invoke a runner?).
I didn't think wrapping these with west would be very useful, so I skipped it.

Reply:
`west flash` invokes a runner, but the runner is written in Python, so it can do anything you can do in Python.

For example the openocd runner: https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/west_commands/runners/openocd.py

Reply:
It seems to me that the most useful thing for something like 'west flash' to target is flashing via the swd interface, i.e. when USB Media access is not an option. I understand that this would normally need openocd. The swd interface should be the same as for any CM0+ chip, so we ought to be able to reuse an implementation from there.
Including an equivalent to "with `cp zephyr.uf2 /media/...`" should be trivial especially as the volume name the pico exposes is fixed.

Reply:
> so we ought to be able to reuse an implementation from there.
Including an equivalent to "with cp zephyr.uf2 /media/..." should be trivial especially as the volume name the pico exposes is fixed.

There's an enhancement request for a usb mass storage runner: https://github.com/zephyrproject-rtos/zephyr/issues/12985

I've unassigned myself since I have no plans on working on it, but that indeed sounds like it would solve the problem of missing west flash support.

In my opinion, we could do that after the initial merge. @yonsch if you want to implement the runner and use it for the RPi Pico in a follow-up PR, I would be very much open to merging the new runner then.